h2/src/main/org/h2/util/DateTimeUtils.java

```diff
@@ -113,13 +113,32 @@ private static Calendar getCalendar() {
     private static Calendar getCalendar(TimeZone tz) {
         Calendar c = CACHED_CALENDAR_NON_DEFAULT_TIMEZONE.get();
         if (c == null || !c.getTimeZone().equals(tz)) {
-            c = Calendar.getInstance(tz);
+            c = DateTimeUtils.createCalendar(tz);
             CACHED_CALENDAR_NON_DEFAULT_TIMEZONE.set(c);
         }
         c.clear();
         return c;
     }
 
+    /**
+     * Creates a calendar for the default timezone.
+     *
```

Comment:
please mention in the comment for this method why this is necessary.

Also please rename to createGregorianCalendar.